### PR TITLE
SW-3908 Changes to PR 1597 (warning dialog on update planting complete on map)

### PR DESCRIPTION
**Pull request by karthikbtf**
- refactored passing of id/val in a more constrained context
- the earlier effort was accidentally merged onto Charlie's branch and reverted again, that PR is no longer valid... sending this out again for Charlie to review

### CODE REVIEW

**Comment by karthikbtf** [comment]
@charlie-tf I left this one for you to review - if you think it makes sense we can get it in otherwise no harm, this doesn't change any functionality.

**Review by charlie-tf — CHANGES_REQUESTED** [review]
Couple small cleanup changes, but otherwise looks good. Thanks for the update!

**Comment by karthikbtf** [comment]
> Couple small cleanup changes, but otherwise looks good. Thanks for the update!

Thanks for the review Charlie, this one should be ready to go as well.

File: src/components/NurseryWithdrawals/PlantingProgressMap.tsx [hunk 1]
```diff
@@ -112,7 +112,7 @@ export default function PlantingProgressMap({ plantingSiteId, reloadTracking }:
 
   const updatePlantingComplete = useCallback(
     (id: number, val: boolean) => {
-      if (!selectedZoneHasStats) {
+      if (val === true || !selectedZoneHasStats) {
```
Comment (charlie-tf) [inline src/components/NurseryWithdrawals/PlantingProgressMap.tsx:115]: I think that in practice this should not happen—there won't be statistics if the subzones aren't already all completed. I don't object to having this check in there though.

**Review by charlie-tf — APPROVED** [review]
👍 Thanks for the changes

File: src/components/NurseryWithdrawals/PlantingProgressMap.tsx (hunk 1 above)
Comment (karthikbtf) [reply src/components/NurseryWithdrawals/PlantingProgressMap.tsx:115]: Ah yes, that is true. Hmm.. I can remove it since it doesn't make sense.

**Review by charlie-tf — APPROVED** [review]

**Pull request merged**